Meta/ladybird.py

```diff
@@ -0,0 +1,295 @@
+        'windows_dev_ninja': build_root_dir / 'debug',
+        'windows_dev_msbuild': build_root_dir / 'debug',
+        'Debug': build_root_dir / 'debug',
+        'Sanitizer': build_root_dir / 'sanitizers',
+        'Distribution': build_root_dir / 'distribution',
+    }
+    if preset not in known_presets:
+        print(f'Unknown build preset "{preset}"', file=sys.stderr)
+        sys.exit(1)
+
+    build_preset_dir = known_presets.get(preset)
+    vcpkg_root = str(build_root_dir / 'vcpkg')
+    os.environ['VCPKG_ROOT'] = vcpkg_root
+    os.environ['PATH'] += os.pathsep + str(lb_source_dir.joinpath('Toolchain', 'Local', 'cmake', 'bin'))
+    os.environ['PATH'] += os.pathsep + str(vcpkg_root)
+    return lb_source_dir, build_preset_dir, cmake_args
+
+
+def _build_vcpkg():
+    sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'Toolchain'))
+    from BuildVcpkg import main as build_vcpkg
+    build_vcpkg()
+
+
+def _validate_cmake_version():
+    import re
+    cmake_pls_install_msg = 'Please install CMake version 3.25 or newer.'
+
+    try:
+        cmake_version_output = subprocess.check_output(
+            [
+                'cmake',
+                '--version',
+            ],
+            text=True
+        ).strip()
+
+        version_match = re.search(r'version\s+(\d+)\.(\d+)\.(\d+)?', cmake_version_output)
+        if version_match:
+            major = int(version_match.group(1))
+            minor = int(version_match.group(2))
+            patch = int(version_match.group(3))
+            if major < 3 or (major == 3 and minor < 25):
+                print(f'CMake version {major}.{minor}.{patch} is too old. {cmake_pls_install_msg}', file=sys.stderr)
+                sys.exit(1)
+        else:
+            print(f'Unable to determine CMake version. {cmake_pls_install_msg}', file=sys.stderr)
+            sys.exit(1)
+    except subprocess.CalledProcessError as e:
+        _print_process_stderr(e, f'CMake not found. {cmake_pls_install_msg}\n')
+        sys.exit(1)
+
+
+def _ensure_ladybird_source_dir():
+    ladybird_source_dir = os.environ.get('LADYBIRD_SOURCE_DIR')
+
+    if not ladybird_source_dir or not Path(ladybird_source_dir).is_dir():
+        try:
+            top_dir = subprocess.check_output(
+                [
+                    'git',
+                    'rev-parse',
+                    '--show-toplevel',
+                ],
+                text=True
+            ).strip()
+
+            ladybird_source_dir = str(Path(top_dir))
+            os.environ['LADYBIRD_SOURCE_DIR'] = ladybird_source_dir
+        except subprocess.CalledProcessError as e:
+            _print_process_stderr(e, 'Unable to determine LADYBIRD_SOURCE_DIR:')
+            sys.exit(1)
+    return Path(ladybird_source_dir)
+
+
+def _build_main(build_dir, **kwargs):
+    build_args = [
+        'cmake',
+        '--build',
+        str(build_dir),
+        '--parallel',
+        str(multiprocessing.cpu_count()),
```

Comment:
the previous logic was this:

```
 # Get either the environment MAKEJOBS or all processors via CMake
 [ -z "$MAKEJOBS" ] && MAKEJOBS=$(cmake -P "$LADYBIRD_SOURCE_DIR/Meta/CMake/processor-count.cmake")
 ```
 which is not equivalent to just calling multiprocessing.cpu_count().
 
 This ignores $MAKEJOBS, which is almost always going to be set to *reduce* build parallelism.